### ISSUE TITLE
[llvm-lit] Show test output when combining -v or -a with -q

### ISSUE THREAD

**Issue by hnrklssn** [llvm/llvm-project#106643]
Lit has various flags to control output, that interact in non-obvious ways when combined: `--quiet --succinct --verbose --show-all`. Their order doesn't matter, they are completely separate flags. 

At the moment, `--quiet` completely overrides `--verbose` and `--show-all`, without any sort of warning. I think it would be more useful if they could be combined such that `--quiet` still silences lit's note and warning diagnostics, but `--verbose` and `--show-all` still can show the output of test cases.

**Comment by jh7370** [comment]
I'm thinking that we might want a more refined "last one wins" style approach, because I think it makes as little sense for -a/-v to produce test output when -q is enabled as the current behaviour. However, regardless, I think this probably needs an RFC for better visibility: the change has the potential to break people's existing scripts etc.

**Comment by hnrklssn** [comment]
To be clear, I specifically want to be able to combine the test output from -a/-v with the diagnostic silencing of -q, not for -a/-v to simply override -q. But fair enough on the RFC part.

**Comment by RoboTux** [comment]
I would agree with James, lit's doc has the following on --quiet:

> Suppress any output except for test failures.

Therefore I would expect --quiet, --succinct, --verbose and --show-all to be mutually exclusive. Last one wins would be compatible with that. I think a separate --mask-warnings (YMMV for naming) which is implied to by --quiet would be better IMO. Where is the RFC by the way? I couldn't find it when searching discourse.

**Comment by RoboTux** [comment]
I agree with --output-format and remapping existing options to it. And sorry I don't have the bandwidth either right now.
